R/dashboard.Rmd

```diff
@@ -1,28 +1,408 @@
+
+```{r animated graphic for top 10 countries by year for exports}
+exp %>% 
+  mutate_if(is.factor, as.character)%>%
+  mutate_if(is.integer, as.numeric) -> exp 
+
+exp_formatted <- exp %>%
+  group_by(Year) %>%
+  mutate(rank = rank(-Value_mill),
+         Value_rel = Value_mill/Value_mill[rank==1],
+         Value_lbl = paste0(" ",round(Value_mill))) %>%
+  group_by(CTY_Name) %>% 
+  filter(rank <=10) %>%
+  ungroup()
+
+
+anim_exp <- ggplot(exp_formatted, 
+                   aes(rank, group = CTY_Name, fill = as.factor(CTY_Name), color = as.factor(CTY_Name))) +
+  
+  geom_tile(aes(y = Value_mill/2, height = Value_mill, width = 0.9),
+            alpha = 0.8, color = NA) +
+  
+  geom_text(aes(y = 0, label = paste(CTY_Name, " ")), vjust = 0.2, hjust = 1) +
+  geom_text(aes(y=Value_mill,label = Value_lbl, hjust=0)) +
+  
+  coord_flip(clip = "off", expand = FALSE) +
+  scale_y_continuous(labels = scales::comma) +
+  scale_x_reverse() +
+  guides(color = FALSE, fill = FALSE) +
+  
+  theme(axis.line=element_blank(),
+        axis.text.x=element_blank(),
+        axis.text.y=element_blank(),
+        axis.ticks=element_blank(),
+        axis.title.x=element_blank(),
+        axis.title.y=element_blank(),
+        legend.position="none",
+        panel.background=element_blank(),
+        panel.border=element_blank(),
+        panel.grid.major=element_blank(),
+        panel.grid.minor=element_blank(),
+        panel.grid.major.x = element_line( size=.1, color="grey" ),
+        panel.grid.minor.x = element_line( size=.1, color="grey" ),
+        plot.title=element_text(size=25, hjust=0.5, face="bold", colour="grey", vjust=-1),
+        plot.subtitle=element_text(size=18, hjust=0.5, face="italic", color="grey"),
+        plot.caption =element_text(size=8, hjust=0.5, face="italic", color="grey"),
+        plot.background=element_blank(),
+        plot.margin = margin(2,2, 2, 4, "cm")) +
+  
+  transition_states(Year, transition_length = 4, state_length = 1, wrap = FALSE) +
+  
+  view_follow(fixed_x = TRUE)  +
+  
+  labs(title = 'Export per Year : {closest_state}',  
+       subtitle  =  "Top 10 Countries Vanuatu has exported to",
+       caption  = "Export Value in Billions of Vatu | Data Source: Asycuda") 
+
+animate(anim_imp, nframes = 100, fps = 5)
+```
+
+
+Export by Commodity
+=====================================
+
+Column {.sidebar}
+------------------------------------------------------------------
+```{r}
+startData <- toptenexport
+GBChoices <- as.list(names(startData))
+names(GBChoices) <- paste(names(startData),map(startData,~length(unique(.x))))
+
+selectInput(inputId = "GB", label = "Group By", choices = GBChoices, selected = "Year 19")
+selectInput(inputId = "Metric", label = "Metric", choices = names(select_if(startData,is.numeric)), selected = "Year")
+
+updateData <- reactive(
+  startData %>% 
+    group_by_(input$GB) %>% 
+    summarise_if(is.numeric, sum, na.rm=T)
+  )

```

Comment:
Change the ordering around `selectInput()` and `reactive` data calls to line up with order of variable use.